CI: use File::Temp to create temporary directory

auerswal opened the pull request:
Before, ci/test-11-unpriv.pl used a hard-coded file name of /tmp/fping.copy to create an unprivileged copy of fping.  This file was not deleted after it was no longer used.

Now, the unprivileged fping copy is created in a temporary directory that is deleted after all tests from the file.

This addresses GH issue #339.

auerswal committed:
CI: use File::Temp to create temporary directory

Before, ci/test-11-unpriv.pl used a hard-coded file name of
/tmp/fping.copy to create an unprivileged copy of fping.  This
file was not deleted after it was no longer used.

Now, the unprivileged fping copy is created in a temporary
directory that is deleted after all tests from the file.

This addresses GH issue #339.

gsnw-sebast commented:
Considering system independence, I would suggest using `File::Copy` instead of `system()`

```
use File::Copy;

copy("datafile.dat", "datafile.bak") or die "copy failed: $!";
```

The chmod function should also be available directly in perl (https://perldoc.perl.org/functions/chmod)

auerswal committed:
CI: use chmod and File::Copy's copy, not system

Before, ci/test-11-unpriv.pl used Perl's "system" function
to execute a Unix command line to copy a file and change the
permissions of the copy.

Now, the Perl functions "copy" (from File::Copy) and "chmod"
are used instead.

gsnw-sebast commented:
You can merge, everything looks good

auerswal commented:
Thanks for the review. I'll go ahead and merge this now.
